python/src/io.cpp

```diff
@@ -232,140 +232,99 @@ void io(py::module& m)
+             instance.write(_points);
            },
-           py::arg("points"),
-           py::arg("encoding") = dolfin::io::XDMFFile::Encoding::HDF5)
+           py::arg("points"))
       // Points with values
       .def("write",
            [](dolfin::io::XDMFFile& instance, py::list points,
-              std::vector<double>& values,
-              dolfin::io::XDMFFile::Encoding encoding) {
+              std::vector<double>& values) {
              auto _points = points.cast<std::vector<dolfin::geometry::Point>>();
-             instance.write(_points, values, encoding);
+             instance.write(_points, values);
            },
-           py::arg("points"), py::arg("values"),
-           py::arg("encoding") = dolfin::io::XDMFFile::Encoding::HDF5)
+           py::arg("points"), py::arg("values"))
       // py:object / dolfin.function.function.Function
       .def("write",
-           [](dolfin::io::XDMFFile& instance, const py::object u,
-              dolfin::io::XDMFFile::Encoding encoding) {
+           [](dolfin::io::XDMFFile& instance, const py::object u) {
              auto _u
                  = u.attr("_cpp_object").cast<dolfin::function::Function*>();
-             instance.write(*_u, encoding);
+             instance.write(*_u);
            },
-           py::arg("u"),
-           py::arg("encoding") = dolfin::io::XDMFFile::Encoding::HDF5)
+           py::arg("u"))
       .def("write",
-           [](dolfin::io::XDMFFile& instance, const py::object u, double t,
-              dolfin::io::XDMFFile::Encoding encoding) {
+           [](dolfin::io::XDMFFile& instance, const py::object u, double t) {
              auto _u
                  = u.attr("_cpp_object").cast<dolfin::function::Function*>();
-             instance.write(*_u, t, encoding);
+             instance.write(*_u, t);
            },
-           py::arg("u"), py::arg("t"),
-           py::arg("encoding") = dolfin::io::XDMFFile::Encoding::HDF5)
+           py::arg("u"), py::arg("t"))
       // Check points
       .def("write_checkpoint",
            [](dolfin::io::XDMFFile& instance,
               const dolfin::function::Function& u, std::string function_name,
-              double time_step, dolfin::io::XDMFFile::Encoding encoding) {
-             instance.write_checkpoint(u, function_name, time_step, encoding);
+              double time_step) {
+             instance.write_checkpoint(u, function_name, time_step);
            },
-           py::arg("u"), py::arg("function_name"), py::arg("time_step") = 0.0,
-           py::arg("encoding") = dolfin::io::XDMFFile::Encoding::HDF5)
+           py::arg("u"), py::arg("function_name"), py::arg("time_step") = 0.0)
       .def("write_checkpoint",
            [](dolfin::io::XDMFFile& instance, const py::object u,
-              std::string function_name, double time_step,
-              dolfin::io::XDMFFile::Encoding encoding) {
+              std::string function_name, double time_step) {
              auto _u
```

Comment:
It would be good to remove all the unwrapping in the pybind11 layer and handle it from the Python interface.